src/types/budget.ts

```diff
@@ -0,0 +1,9 @@
+import { Types } from "mongoose";
+import { ICategory } from "./category";
+
+export type IBudget = {
+  user: Types.ObjectId;
```

Comment:
should also have an optional `_id` field, since mongo will create it, may be required for further CRUD ops.